src/database/prepare.ts

```diff
@@ -0,0 +1,36 @@
+// Before importing any other module,
+// loads all the environment variables from a .env file
+// @see https://github.com/motdotla/dotenv/tree/v6.1.0#how-do-i-use-dotenv-with-import
+import * as dotenv from "dotenv";
+dotenv.config();
+
+import * as path from "path";
+import { Sequelize } from "sequelize";
+import * as usync from "umzug-sync";
+import { populateIpaPublicAdministrationTable } from "../services/ipaPublicAdministrationService";
+import { log } from "../utils/logger";
+import sequelize from "./db";
+
+/**
+ * Apply migrations to db and populate the table of public administrations from IPA
+ */
+usync
+  .migrate({
+    SequelizeImport: Sequelize,
+    logging: (param: string) => log.info("%s", param),
+    migrationsDir: path.join("dist", "migrations"),
+    sequelize
+  })
+  .catch(error => {
+    log.error("Failed to apply migrations. %s", error);
+    process.exit(1);
+  })
+  .then(() => populateIpaPublicAdministrationTable())
```

Comment:
don't do this here, this is not the responsibility of a web application

Reply:
I wrote this line here because the table of the public administrations should be already populated when the app starts, so this function should be called after the migrations are applied and before the app is running and listening. If it shouldn't be done here, where else should it be done and whose responsibility is this?

Reply:
the app should handle the scenario where the table is not populated (i.e. exit if empty, or disable the affected functionality until the table is filled, or just do nothing?)

a separate process, run via [k8s job](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) is responsible for keeping this table up to date - the code of such script should be part of this application, it's just executed separately

Reply:
I'm sorry, this is still a little unclear to me. As far as I understood:
- the code of the updating process should be moved into a separate file so that it will be executed via k8s job.
- the populating process, which is expected to be run only once, should not be taken care of by the app, so it can be removed from the app.

The updating process should be scheduled to run at night by k8s job; but when the table will be populated for the first time and by whom?

Reply:
what's the difference between the first population step and the updates? they both rely on calling `upsertFromIpa` - anyway, keep the logic as is for now, we may improve it later